### PR TITLE
chore(gatsby-plugin-mdx): refactor plugin fetcher

### PR DESCRIPTION
**Pull request by pvdz**
Small refactor
- eliminate redundant arrow wrapping
- use one array to push plugins onto
- map them to `[func, {}]` at the end
- replaced `_.isFunction` with plain `typeof x !== 'function'` and dropped the lodash dep

Some stuff unaddressed that I wonder about:

- I'm assuming that the array is even required. For example, babel will also just accept a function there. At the same time, we could add the name/origin of the function in here which might help with debugging. I didn't change it either way.

- ~I'm not sure whether the `interopDefault` wrap is needed when requiring so I left it, but I feel that's no longer required.~

- I wonder whether ``fileNodes = getNodesByType(`File`)`` is a bug. That depends on when this is actually called, whether more file nodes are added after this fact, and more importantly whether the compiler is called once that happens. Did not change that logic in this PR.

### CODE REVIEW

**Comment by pieh** [comment]
> * I'm not sure whether the `interopDefault` wrap is needed when requiring so I left it, but I feel that's no longer required.

We don't control all plugins that can be imported there so I would keep the `interopDefault` - this seems like breaking change

**Comment by pvdz** [comment]
Yeah wasn't going to remove it. Just wondering. But what you're saying makes sense.

File: packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js [hunk 1]
```diff
@@ -20,66 +19,56 @@ module.exports = async function getSourcePluginsAsRemarkPlugins({
...
-              reporter,
-              cache,
-              ...helpers,
-            },
-            plugin.options || {}
-          )
+  gatsbyRemarkPlugins.forEach(plugin => {
+    const importedPlugin = interopDefault(require(plugin.resolve))
+    if (typeof importedPlugin !== `function`) {
+      debug(`userPlugins: filtering out`, plugin)
```
Comment (wardpeet) [inline packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js:47]:
```suggestion
      debug(`userPlugins: not a valid plugin`, plugin)
```

File: packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js [hunk 2]
```diff
@@ -20,66 +19,56 @@ module.exports = async function getSourcePluginsAsRemarkPlugins({
   ...helpers
 }) {
   debug(`getSourcePluginsAsRemarkPlugins`)
-  let pathPlugin = undefined
+
+  let userPlugins = []
```
Comment (wardpeet) [inline packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js:23]: can't this be const?
```suggestion
  const userPlugins = []
```

File: packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js (hunk 1 above)
Comment (pvdz) [reply packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js:47]: Not my change, but let's update it anyways :)

File: packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js (hunk 2 above)
Comment (pvdz) [reply packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js:23]: Ah yes

File: packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js [hunk 3]
```diff
@@ -20,66 +19,56 @@ module.exports = async function getSourcePluginsAsRemarkPlugins({
...
-              reporter,
-              cache,
-              ...helpers,
-            },
-            plugin.options || {}
-          )
+  gatsbyRemarkPlugins.forEach(plugin => {
+    const importedPlugin = interopDefault(require(plugin.resolve))
+    if (typeof importedPlugin !== `function`) {
+      debug(`userPlugins: not a valid plugin: \`${plugin}\``)
```
Comment (wardpeet) [inline packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js:47]: Wouldn't this result in a "[object Object]" scenario? as plugin is an object?

Comment (pvdz) [reply packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js:47]: Hm yeah you're probably right :)

**Comment by pvdz** [comment]
I'll revisit this later. I realized that we should try to make a habbit of collecting all the api's and keep them boxed in one arg, directly accessing them. Something like `gatsbyApi` and `gatsbyApi.getNodes()`, rather than unbox them and randomly putting the remainder in `helpers`. So I'm just gonna close this.

**Pull request closed without merging**